File: python/chapter_2/selection_sort.py
def find_smallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index

def selection_sort(arr):
    res = []
    for i in range(len(arr)):
        smallest = find_smallest(arr)
        res.append(arr.pop(smallest))
    return res

data = input('Введите элементы массива через пробел: ')
data = list(map(int, data.split()))
print(selection_sort(data))

File: go/chapter2/selection_sort.go
package selection_sort

func findSmallest(arr []int) int {
	s := arr[0]
	si := 0
	for i, v := range arr {
		if v < s {
			s = v
			si = i
		}
	}
	return si
}

func SelectionSort(arr []int) ([]int) {
	res := make([]int, len(arr))
	if len(arr) == 0 {
		return res
	}
	l := len(arr)
	var s int
	for i := 0; i < l; i++ {
		s = findSmallest(arr)
		res[i] = arr[s]
		arr = append(arr[:s], arr[s+1:]...)
	}
	return res
}